public/app/features/dashboard-scene/serialization/transformSceneToSaveModelSchemaV2.ts

```diff
@@ -401,18 +405,22 @@ function getAnnotations(state: DashboardSceneState): AnnotationQueryKind[] {
         datasource: layer.state.query.datasource || getDefaultDataSourceRef(),
         query: {
           kind: getAnnotationQueryKind(layer.state.query),
-          spec: omit(layer.state.query, 'datasource'),
+          // layer.state.query returns an object which is AnnotationQueryKind
+          // with the query field, and inside that field we'll have
+          // this AnnotationQuery object and the things we care about are inside the query.spec field
+          spec: layer.state.query.query.spec,
```

Comment:
I'm open to suggestions here, because this is not ideal, but in v2 where are passing the AnnotationQuery spec to scenes `layer.state.query` field, which then means that on transforming from scene to v2: we get: 
```
layer: {
    state: {
        query: {
            ...other AnnotationQuerySpec fields
            query: {
                kind: 
                spec: DataQueryKind --> this is what we need in the query, not the whole AnnotationQuery object coming from the scene
            }
    }
}
```